reconcilers/flow_test.go

```diff
@@ -269,6 +271,132 @@ func TestWhile(t *testing.T) {
 	})
 }
 
+func TestForEach(t *testing.T) {
+	testNamespace := "test-namespace"
+	testName := "test-resource"
+
+	scheme := runtime.NewScheme()
+	_ = resources.AddToScheme(scheme)
+
+	resource := dies.TestResourceBlank.
+		MetadataDie(func(d *diemetav1.ObjectMetaDie) {
+			d.Namespace(testNamespace)
+			d.Name(testName)
+		}).
+		SpecDie(func(d *dies.TestResourceSpecDie) {
+			d.Fields(map[string]string{})
+		})
+
```

Comment:
**[thought, off-topic]:** In terms of reading order _this_ is the moment where I expect to see the reconciler-under-test. I keep scrolling back and forth between the test cases and the reconciler below. It's absolutely fine. Declaring the controller here would work, but it's not something we do conventionally in our test suite. All good 😄

Reply:
there are places were we do define a unique reconciler for each test case. Since there are only a couple test cases I think this is cleaner, but if those cases needed to diverge more then it would be worth localizing the reconciler definition.